redisgears_core/src/function_load_command.rs

```diff
@@ -132,17 +99,22 @@ pub(crate) fn function_load_intrernal(
     let backend = backend.unwrap();
     let compile_lib_ctx = CompiledLibraryAPI::new();
     let compile_lib_internals = compile_lib_ctx.take_internals();
-    let lib_ctx =
-        backend.compile_library(code, meta_data.config.as_ref(), Box::new(compile_lib_ctx));
+    let lib_ctx = backend.compile_library(
+        &meta_data.name,
+        code,
+        meta_data.api_version,
+        meta_data.config.as_ref(),
+        Box::new(compile_lib_ctx),
+    );
     let lib_ctx = match lib_ctx {
-        Err(e) => return Err(format!("Failed library compilation {}", e.get_msg())),
+        Err(e) => return Err(format!("Failed library compilation:\n{}", e.get_msg())),
```

Comment:
I believe we should avoid adding `\n` here, this error will pass the the user eventually and RESP does not support new lines on errors anyway and it could brake the protocol (`redismodule-rs` will protect us from this and will replace the `\n` with space IIRC)

Reply:
Please notice that on resp3 we have error blob type that can contain `\n`. We can leverage that for more complex errors like JS traces for example ...